vowpalwabbit/parse_example.cc

```diff
@@ -148,20 +170,63 @@ class TC_parser
     {
       // maybeFeature --> 'String' FeatureValue
       VW::string_view feature_name = read_name();
-      _v = _cur_channel_v * featureValue();
+      VW::string_view string_feature_value;
+
+      float float_feature_value = 0.f;
+      bool is_feature_float = isFeatureValueFloat(float_feature_value);
+
+      if (_chain_hash)
+      {
+        if (is_feature_float)
```

Comment:
[Minor] Consider unifying conditional into the root if, to avoid duplicating `_v = _cur_channel_v * float_feature_value;`.